fieldpath/set.go

```diff
@@ -676,6 +705,7 @@ func (s *SetNodeMap) RecursiveDifference(s2 *Set) *SetNodeMap {
 }
 
 // EnsureNamedFieldsAreMembers returns a set that contains all the named fields along with the leaves.
+// Missing keys in list maps are backfilled with null values.
```

Comment:
I really don't think we should add explicit null values for missing keys, but should fix comparisons instead to treat non-identical key sets as not equal. That will probably require changing the docs and maybe name of this method to something more like `EnsureExistingNamedFieldsAreMembers` and making sure callers are ok with subsets of the named fields